examples/multiple-datasets-with-headers/app.tsx

```diff
@@ -0,0 +1,75 @@
+/** @jsx h */
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createQuerySuggestionsPlugin } from '@algolia/autocomplete-plugin-query-suggestions';
+import { createLocalStorageRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+import algoliasearch from 'algoliasearch';
+import { h, Fragment } from 'preact';
+
+import '@algolia/autocomplete-theme-classic';
+import { createCategoriesPlugin } from './categoriesPlugin';
+
+const appId = 'latency';
+const apiKey = '6be0576ff61c053d5f9a3225e2a90f76';
+const searchClient = algoliasearch(appId, apiKey);
+
+const recentSearchesPlugin = createLocalStorageRecentSearchesPlugin({
+  key: 'qs-with-rs-example',
+  limit: 3,
+  transformSource({ source }) {
+    return {
+      ...source,
+      templates: {
+        ...source.templates,
+        header({ items, state }) {
+          if (Boolean(state.query) || items.length === 0) {
```

Comment:
Isn't this check enough? (same below)

```suggestion
          if (items.length === 0) {
```